kafka/common.go

```diff
@@ -209,3 +270,21 @@ func (cfg *CommonConfig) newClient(additionalOpts ...kgo.Opt) (*kgo.Client, erro
 	client.ForceMetadataRefresh()
 	return client, nil
 }
+
+func newAWSMSKIAMSASL() (sasl.Mechanism, error) {
+	awscfg, err := awsconfig.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("kafka: error loading AWS config: %w", err)
+	}
```

Comment:
Good question, which we should have a test for :)
I thought the Retrieve method would reload the credentials, but maybe it doesn't. I'll check this tomorrow, and either way I'll add a test.

Reply:
I played with it a bit, and ended up looking at the aws-sdk code -- there's no validation of those env vars AFAICT. If they're non-empty, then they're used, otherwise the SDK goes onto the next option -- credentials file.

There is validation of the credentials file, but I figure that an invalid credentials file should be treated the same as an outdated credentials file, and reloaded upon being fixed. If you disagree with that, let me know and we can come back to it.